### DISCUSSION TITLE
Ambiq RTC differences

### DISCUSSION THREAD

**Discussion by RichardSWheatley** [zephyrproject-rtos/zephyr#72480] [Q&A] [answered]
Hello, @bjarki-trackunit!

You are the maintainer for the RTC section.
I am writing the Ambiq RTC driver when I noticed that the tests for RTC make some assumptions that are not true for the Ambiq RTC.

1. We do not support 99-00 rollover which I already fixed in the tests by adding a check for EINVAL.
```
	if (ret == -EINVAL) {
		TC_PRINT("Rollover not supported\n");
		ztest_test_skip();
	} else {
		zassert_ok(ret, "RTC Set Time Failed");
	}
```
2. When setting the alarms, for the Ambiq RTC you must set the entire struct with a non-zero value

This makes our implementation fail the most of the testing without additions to the tests.
I do not like to make additions to tests unless it is absolutely needed.

Do you think there is a better way to accomplish this?

![image](https://github.com/zephyrproject-rtos/zephyr/assets/12254017/514f19be-78ee-44c3-a5aa-bd8c6971cafc)
![image](https://github.com/zephyrproject-rtos/zephyr/assets/12254017/b45e60c5-edd0-4b38-aa72-707b34d08fbc)

**Comment by bjarki-andreasen** [comment, marked as answer]
Regarding issue 1, returning -EINVAL is the correct behavior, and skipping the test if -EINVAL is returned is appropriate.

Regarding issue 2, the test suite can be updated to set every supported field, rather than just hours and minutes. This is a valid configuration which should be supported by all RTCs.

We can add an additional test to validate that only the selected alarm time fields are used, which is a valid configuration for most RTCs, this test can be skipped if the RTC returns -EINVAL, which would be the case for the AMBIQ RTC.

**Reply by RichardSWheatley** [reply]
Thank you! I will create a draft pull request for this and you can check my implementation.
